pkg/k8s/watchers/cilium_endpoint.go

```diff
@@ -235,3 +253,59 @@ func (k *K8sWatcher) endpointDeleted(endpoint *types.CiliumEndpoint) {
 		k.egressGatewayManager.OnDeleteEndpoint(endpoint)
 	}
 }
+
+// ciliumEndpointManager manages tasks related to ciliumendpoints, including
+// performing periodic cleanup of local stale CEP resources.
+type ciliumEndpointManager struct {
+	*lock.RWMutex
+	ciliumEndpointsCleanupEnabled bool
```

Comment:
I have this for two reasons:

Wanted to ensure that cleanStaleCEPs was done before the GC could run on this (i.e. someone could create a race if they set the gc interval to be very small)

As well, In the case that gc-interval is set to 0, and the ongoing gc is disabled, I wanted to avoid the k8s watcher potentially indefinitely appending to the marked list that never gets sweeped.